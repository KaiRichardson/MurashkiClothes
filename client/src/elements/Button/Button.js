import React from 'react';
import styled, { css } from 'styled-components';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';

import { spacing, transition, white, black, grey } from 'utils';

/*
    @desc Button component.
        Add optional dark prop for dark button, default is light.
*/

export const Button = styled.button`
  --bg: ${white};
  --txtColor: ${black};

  cursor: pointer;
  padding: ${spacing.sm} ${spacing.md};

  /* Default Button */
  background: var(--bg);
  color: var(--txtColor);
  border: none;
  ${transition({})};

  &:hover {
    --bg: ${black};
    --txtColor: ${white};
  }

  /* Dark Button */
  ${props =>
    props.dark &&
    css`
      --bg: ${black};
      --txtColor: ${white};

      &:hover {
        --bg: ${grey};
      }
    `}
`;

export const ButtonLink = styled(({ dark, ...rest }) => <Link {...rest} />)`
  /* Default Button */
  --bg: ${white};
  --txtColor: ${black};

  background: var(--bg);
  color: var(--txtColor);

  border: none;
  text-decoration: none;
  ${transition({})};
  cursor: pointer;
  padding: ${spacing.sm} ${spacing.md};

  &:hover {
    --bg: ${black};
    --txtColor: ${white};
  }

  /* Dark Button */
  ${props =>
    props.dark &&
    css`
      --bg: ${black};
      --txtColor: ${white};

      &:hover {
        --bg: ${grey};
      }
    `}
`;

Button.propTypes = {
  dark: PropTypes.bool
};

ButtonLink.propTypes = {
  dark: PropTypes.bool
};
