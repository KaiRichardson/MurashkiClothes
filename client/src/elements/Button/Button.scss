@import '../../assets/sass/main.scss';

.btn {
  --bg: var(--clr-white);
  --txtColor: var(--clr-black);

  padding: $spacing(sm) $spacing(md);

  /* Default Button */
  background: var(--bg);
  color: var(--txtColor);

  &:hover {
    --bg: var(--clr-black);
    --txtColor: var(--clr-white);
  }

  /* Dark Button */
  &--dark {
    --bg: var(--clr-black);
    --txtColor: var(--clr-white);

    &:hover {
      --bg: var(--clr-grey);
    }
  }
}
