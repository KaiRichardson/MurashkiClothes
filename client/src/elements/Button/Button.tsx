import React from 'react';
import styled, { css } from 'styled-components';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';

import { spacing, transition, white, black, grey, lightGrey, red } from 'utils';

/*
    @desc Button component.
        Add optional dark prop for dark button, default is light.
*/

export const Button = styled.button<{ dark?: boolean; trans?: boolean; brand?: boolean }>`
  --bg: ${white};
  --txtColor: ${black};

  cursor: pointer;
  padding: ${spacing.sm} ${spacing.md};
  border: none;
  ${transition({})};

  /* Default Button */
  background: var(--bg);
  color: var(--txtColor);

  &:hover {
    --bg: ${black};
    --txtColor: ${white};
  }

  /* Dark Button */
  ${props =>
    props.dark &&
    css`
      --bg: ${black};
      --txtColor: ${white};

      &:hover {
        --bg: ${grey};
      }
    `}

    /* Brand Color Button */
    ${props =>
      props.brand &&
      css`
        --bg: ${black};
        --txtColor: ${white};

        &:hover {
          --bg: ${red};
        }
      `}

    /* Transparent Button */
    ${props =>
      props.trans &&
      css`
        --txtColor: ${lightGrey};
        --bg: transparent;

        &:hover {
          --txtColor: ${black};
          --bg: transparent;
        }
      `}
`;

export const ButtonLink = styled(({ dark, trans, brand, ...rest }) => <Link {...rest} />)<{
  dark?: boolean;
  trans?: boolean;
  brand?: boolean;
}>`
  /* Default Button */
  --bg: ${white};
  --txtColor: ${black};

  background: var(--bg);
  color: var(--txtColor);

  text-align: center;
  display: inline-block;
  border: none;
  text-decoration: none;
  ${transition({})};
  cursor: pointer;
  padding: ${spacing.sm} ${spacing.md};
  

  &:hover {
    --bg: ${black};
    --txtColor: ${white};
  }

  /* Dark Button */
  ${props =>
    props.dark &&
    css`
      --bg: ${black};
      --txtColor: ${white};

      &:hover {
        --bg: ${grey};
      }
    `}

    /* Brand Color Button */
    ${props =>
      props.brand &&
      css`
        --bg: ${black};
        --txtColor: ${white};

        &:hover {
          --bg: ${red};
        }
      `}

  /* Transparent Button */
  ${props =>
    props.trans &&
    css`
      --txtColor: ${lightGrey};
      --bg: transparent;

      &:hover {
        --txtColor: ${black};
        --bg: transparent;
      }
    `}
`;

Button.propTypes = {
  dark: PropTypes.bool,
  trans: PropTypes.bool
};

ButtonLink.propTypes = {
  dark: PropTypes.bool,
  trans: PropTypes.bool
};
