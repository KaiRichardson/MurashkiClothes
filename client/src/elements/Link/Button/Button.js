import styled, { css } from 'styled-components';
import PropTypes from 'prop-types';

import { spacing, transition, white, black, grey } from './node_modules/utils';

/*
    @desc Button component.
        Add optional dark prop for dark button, default is light.
*/

export const Button = styled.button`
  --bg: ${white};
  --txtColor: ${black};

  padding: ${spacing.sm} ${spacing.md};

  /* Default Button */
  background: var(--bg);
  color: var(--txtColor);
  border: none;
  ${transition({})};

  &:hover {
    --bg: ${black};
    --txtColor: ${white};
  }

  /* Dark Button */
  ${props =>
    props.dark &&
    css`
      --bg: ${black};
      --txtColor: ${white};

      &:hover {
        --bg: ${grey};
      }
    `}
`;

Button.propTypes = {
  dark: PropTypes.bool
};
