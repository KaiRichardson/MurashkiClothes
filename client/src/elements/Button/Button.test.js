import React from 'react';
import { render } from '@testing-library/react';

import Button from './Button';

describe('<Button /> tests', () => {
  it('renders a default button', () => {
    //* Act
    const { getByTestId } = render(<Button data-testid='testButton'>Test Button</Button>);

    //* Assert
    const testButton = getByTestId('testButton');

    expect(testButton.classList[0]).toBe('btn');
  });

  it('renders a dark button', () => {
    //* Act
    const { getByTestId } = render(
      <Button dark data-testid='testButton'>
        Test Button
      </Button>
    );

    //* Assert
    const testButton = getByTestId('testButton');

    expect(testButton.classList[0]).toBe('btn--dark');
  });
});
