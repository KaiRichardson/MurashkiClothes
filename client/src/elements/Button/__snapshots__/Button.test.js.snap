// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> tests renders a dark button 1`] = `
.c0 {
  --bg: #fff;
  --txtColor: #000;
  cursor: pointer;
  padding: 0.5rem 1rem;
  background: var(--bg);
  color: var(--txtColor);
  border: none;
  -webkit-transition: all 250ms ease-in;
  transition: all 250ms ease-in;
  --bg: #000;
  --txtColor: #fff;
}

.c0:hover {
  --bg: #000;
  --txtColor: #fff;
}

.c0:hover {
  --bg: #333;
}

<button
  class="c0"
  data-testid="testButton"
>
  Test Button
</button>
`;

exports[`<Button /> tests renders a default button 1`] = `
.c0 {
  --bg: #fff;
  --txtColor: #000;
  cursor: pointer;
  padding: 0.5rem 1rem;
  background: var(--bg);
  color: var(--txtColor);
  border: none;
  -webkit-transition: all 250ms ease-in;
  transition: all 250ms ease-in;
}

.c0:hover {
  --bg: #000;
  --txtColor: #fff;
}

<button
  class="c0"
  data-testid="testButton"
>
  Test Button
</button>
`;

exports[`<ButtonLink /> tests renders a dark button link 1`] = `
.c0 {
  --bg: #fff;
  --txtColor: #000;
  background: var(--bg);
  color: var(--txtColor);
  border: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: all 250ms ease-in;
  transition: all 250ms ease-in;
  cursor: pointer;
  padding: 0.5rem 1rem;
  --bg: #000;
  --txtColor: #fff;
}

.c0:hover {
  --bg: #000;
  --txtColor: #fff;
}

.c0:hover {
  --bg: #333;
}

<a
  class="c0"
  data-testid="testButton"
  href="/test"
>
  test link
</a>
`;

exports[`<ButtonLink /> tests renders a default button link 1`] = `
.c0 {
  --bg: #fff;
  --txtColor: #000;
  background: var(--bg);
  color: var(--txtColor);
  border: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: all 250ms ease-in;
  transition: all 250ms ease-in;
  cursor: pointer;
  padding: 0.5rem 1rem;
}

.c0:hover {
  --bg: #000;
  --txtColor: #fff;
}

<a
  class="c0"
  data-testid="testButton"
  href="/test"
>
  test link
</a>
`;
